go/mysql/server.go

```diff
@@ -591,7 +592,7 @@ func (c *Conn) writeHandshakeV10(serverVersion string, authServer AuthServer, en
 	pos = writeUint16(data, pos, uint16(capabilities))
 
 	// Character set.
-	pos = writeByte(data, pos, CharacterSetUtf8)
+	pos = writeByte(data, pos, CharacterSetUtf8mb4)
```

Comment:
`CharacterSetUtf8mb4` is set to `45` which is `uf8mb4_general_ci`, shouldn't we change that constant to use `utf8mb4_0900_ai_ci` (use `collations.DefaultConnectionCharset`)?

Reply:
It got resolved via https://github.com/vitessio/vitess/commit/8b26ffe088966a8e8cfe0c1fd1b1b22ed5f21940 :)